src/bun.js/bindings/JSEnvironmentVariableMap.cpp

```diff
@@ -125,19 +135,30 @@ JSValue createEnvironmentVariablesMap(Zig::GlobalObject* globalObject)
         object = constructEmptyObject(globalObject, globalObject->objectPrototype());
     }
 
+#if OS(WINDOWS)
+    JSArray* keyArray = constructEmptyArray(globalObject, nullptr, count);
+#endif
+
     static NeverDestroyed<String> TZ = MAKE_STATIC_STRING_IMPL("TZ");
     bool hasTZ = false;
     for (size_t i = 0; i < count; i++) {
         unsigned char* chars;
         size_t len = Bun__getEnvKey(list, i, &chars);
         auto name = String::fromUTF8(chars, len);
+#if OS(WINDOWS)
+        keyArray->putByIndexInline(globalObject, (unsigned)i, jsString(vm, name), false);
+#endif
         if (name == TZ) {
             hasTZ = true;
             continue;
         }
         ASSERT(len > 0);
-
-        Identifier identifier = Identifier::fromString(vm, name);
+#if OS(WINDOWS)
+        String idName = name.convertToASCIIUppercase();
```

Comment:
do we need to keep the original casing as well? `Object.keys` on `process.env` will report the original

Reply:
yes due to the proxy ownKeys